docs/data/material/components/material-icons/SearchIcons.js

```diff
@@ -518,40 +540,110 @@ const Input = styled(InputBase)({
   flex: 1,
 });
 
-const searchIndex = new FlexSearchIndex({
-  tokenize: 'full',
-});
-
 const allIconsMap = {};
+const themeRegEx = /(Outlined|Rounded|TwoTone|Sharp)$/g;
+
 const allIcons = Object.keys(mui)
-  .sort()
+  .sort() // Show ASC
   .map((importName) => {
     let theme = 'Filled';
     let name = importName;
 
-    for (const currentTheme of ['Outlined', 'Rounded', 'TwoTone', 'Sharp']) {
-      if (importName.endsWith(currentTheme)) {
-        theme = currentTheme === 'TwoTone' ? 'Two tone' : currentTheme;
-        name = importName.slice(0, -currentTheme.length);
-        break;
-      }
+    const matchTheme = importName.match(themeRegEx);
+    if (matchTheme !== null) {
+      theme = matchTheme[0] === 'TwoTone' ? 'Two tone' : matchTheme[0];
+      name = importName.slice(0, -matchTheme[0].length);
     }
     let searchable = name;
     if (synonyms[searchable]) {
       searchable += ` ${synonyms[searchable]}`;
     }
-    searchIndex.add(importName, searchable);
 
     const icon = {
+      id: importName, // used by miniSearch
       importName,
       name,
       theme,
+      searchable,
       Component: mui[importName],
     };
     allIconsMap[importName] = icon;
     return icon;
   });
 
+function addSuffixes(term, minLength) {
+  if (term == null) {
+    return undefined;
+  }
+
+  const tokens = [term.toLowerCase()];
+
+  for (let i = 0; i <= term.length - minLength; i += 1) {
+    tokens.push(term.slice(i).toLowerCase());
+  }
+  return tokens;
+}
+
+const miniSearch = new MiniSearch({
+  fields: ['searchable'], // fields to index for full-text search
+  processTerm: (term) => addSuffixes(term, 4),
+  storeFields: ['name', 'Component'],
+  searchOptions: {
+    processTerm: MiniSearch.getDefault('processTerm'),
+    prefix: true,
+    fuzzy: 0.1, // Allow some typo
+    boostDocument: (documentId, term, storedFields) => {
+      // Show exact match first
+      return term.toLowerCase() === storedFields.name.toLowerCase() ? 2 : 1;
+    },
+  },
+});
+
+// Copied from mui-x/packages/x-data-grid-generator/src/services/asyncWorker.ts
+// https://lucaong.github.io/minisearch/classes/MiniSearch.MiniSearch.html#addAllAsync is crap.
+function asyncWorker({ work, tasks, done }) {
```

Comment:
Could this be simplified with [`scheduler.yield`](https://developer.mozilla.org/en-US/docs/Web/API/Scheduler/yield#basic_usage)?